src/CoreBundle/Service/MaintenanceModeService.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace ChameleonSystem\CoreBundle\Service;
+
+use ChameleonSystem\CoreBundle\Exception\MaintenanceModeErrorException;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+
+class MaintenanceModeService implements MaintenanceModeServiceInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function isActivated(): bool
+    {
+        return file_exists(PATH_MAINTENANCE_MODE_MARKER);
+    }
+
+    public function isActivatedInDb(): bool
```

Comment:
How about something like "isRequestedActive"? I'd like to avoid the interface exposing a) which storage type is used and b) that there even are different storage types.

Reply:
see below